File: backend/models/Payment.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Payment = sequelize.define('Payment', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  purchase_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'purchases',
      key: 'id'
    }
  },
  payment_system: {
    type: DataTypes.ENUM('click', 'payme', 'card'),
    allowNull: false
  },
  transaction_id: {
    type: DataTypes.STRING(255),
    allowNull: true,
    unique: true
  },
  amount: {
    type: DataTypes.INTEGER,
    allowNull: false,
    comment: 'Сумма в сумах'
  },
  status: {
    type: DataTypes.ENUM('pending', 'processing', 'completed', 'failed', 'cancelled'),
    defaultValue: 'pending'
  },
  payment_url: {
    type: DataTypes.TEXT,
    allowNull: true,
    comment: 'URL для перенаправления на оплату'
  },
  callback_data: {
    type: DataTypes.JSONB,
    allowNull: true,
    comment: 'Данные от платежной системы'
  },
  error_message: {
    type: DataTypes.TEXT,
    allowNull: true
  }
}, {
  tableName: 'payments',
  timestamps: true
});

module.exports = Payment;

File: backend/app/models/payment.py
from datetime import datetime, timezone

from app import db


class Payment(db.Model):
    __tablename__ = "payments"

    id = db.Column(db.Integer, primary_key=True)
    purchase_id = db.Column(
        db.Integer, db.ForeignKey("purchases.id"), nullable=False
    )
    payment_system = db.Column(db.String(50), nullable=False)
    transaction_id = db.Column(db.String(255), unique=True, nullable=True)
    amount = db.Column(db.Integer, nullable=False)
    status = db.Column(db.String(50), default="pending")
    payment_url = db.Column(db.Text, nullable=True)
    callback_data = db.Column(db.JSON, nullable=True)
    created_at = db.Column(
        db.DateTime, default=lambda: datetime.now(timezone.utc)
    )

    def to_dict(self):
        return {
            "id": self.id,
            "purchase_id": self.purchase_id,
            "payment_system": self.payment_system,
            "transaction_id": self.transaction_id,
            "amount": self.amount,
            "status": self.status,
            "payment_url": self.payment_url,
            "created_at": self.created_at.isoformat() if self.created_at else None,
        }
